### ISSUE TITLE
cache path

### ISSUE THREAD

**Issue by 532910** [vector-im/riot-web#11066]
Cache must be in `~/.cache` not in `.config/Riot/Cache`

**Comment by t3chguy** [comment]
This is the default location for all electron apps: https://github.com/electron/electron/blob/master/docs/api/app.md#appsetpathname-path

**Comment by 532910** [comment]
https://github.com/electron/electron/issues/8124
3 years! f.....g electron

https://standards.freedesktop.org/basedir-spec/basedir-spec-latest.html

**Comment by 532910** [comment]
Look like riot should workaround this with `app.setPath("cache", "~/.cache")`

**Comment by foresto** [comment]
Indeed, Riot creates at least three cache directories in ~/.config/Riot: Cache, Code Cache, and GPUCache. These should be under ~/.cache/Riot instead. They're wasting space in my backup archives.

**Comment by foresto** [comment]
Indeed, Riot creates at least three cache directories in ~/.config/Riot: Cache, Code Cache, and GPUCache. These should be under ~/.cache/Riot instead. They're wasting space in my backup archives, and circumventing the tmpfs I use for cache filesystems.

**Comment by 532910** [comment]
what is the problem to workaround the path with `app.setPath("cache", "~/.cache")` ?

**Comment by 532910** [comment]
Is there any update?

**Comment by t3chguy** [comment]
No, otherwise you would have seen it here.

**Comment by 532910** [comment]
what's wrong to workaround with `app.setPath`?

**Comment by t3chguy** [comment]
Best off asking the Electron team that, they must have a reason for not using it.
I'd imagine `~/.cache` isn't available on absolutely every single supported Linux environment.

**Comment by 532910** [comment]
`.cache` and `.config` are both from the same specification:
https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html

**Comment by t3chguy** [comment]
Doesn't mean that `.config` isn't also specified by some other standard.
As I already said, this would be best done upstream.

**Comment by t3chguy** [comment]
According to https://github.com/electron/electron/issues/8124 setting `cache` dir won't fix it anyway:

> Cache data is written to `userData` dir instead of `cache` dir

**Comment by DC7IA** [comment]
> This is the default location for all electron apps

Just because they don't read standards does it has to be reproduced.

$XDG_CONFIG_HOME is supposed to be a folder you can simply copy to new machines or sync to other machines.

This likely breaks badly if I try to sync with other machines.

**Comment by rugk** [comment]
Great! Though the PR says it "allow[s] customizing browser data location" and goes on to explain how one (i.e. electron dev) can actually use that. As such, I guess as for Element there is still something to do, i.e. actually configure it to use the "correct" locations… (still wonder why they don't use/include the proper defaults then in Electron upstream?)
